### PR TITLE
Fix type of `Nullable.t` which was not untagged in the implementation.

### PR DESCRIPTION
**Pull request by cristianoc**
See https://github.com/rescript-lang/rescript-compiler/pull/6669#issuecomment-2165347349

### CODE REVIEW

File: src/intl/Core__Intl__Common.res [hunk 1]
```diff
@@ -1,4 +1,4 @@
-type localeMatcher = [#lookup | @as("best fit") #bestFit]
+type localeMatcher = [#lookup | #bestFit]
```
Comment (glennsl) [inline src/intl/Core__Intl__Common.res:1]: This still seems wrong. It should be `#"best fit"`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#localematcher

**Review by zth — APPROVED** [review]
Great! Want a release with this right away?

**Comment by cristianoc** [comment]
> Great! Want a release with this right away?

That would unblock the compiler fix.

**Comment by zth** [comment]
@cristianoc great, releasing!

**Comment by zth** [comment]
@cristianoc out in `1.5.0`.

**Pull request merged**